spec/system/consumer/shopping/products_spec.rb

```diff
@@ -82,5 +93,59 @@
         end
       end
     end
+
+    describe "filtering" do
+      before do
+        exchange1.update_attribute :pickup_time, "monday"
+        add_variant_to_order_cycle(exchange1, variant)
+        add_variant_to_order_cycle(exchange1, variant2)
+      end
+
+      context "product taxonomies" do
+        before do
+          login_as user
+          visit edit_admin_enterprise_path(distributor)
+
+          within(".side_menu") do
+            click_link "Shop Preferences"
+          end
+          find(".select2-input").click
+          find(".select2-result-label", text: "Tricky Taxon").click
+          click_button 'Update'
+          expect(flash_message).to eq('Enterprise "Testing Farm" has been successfully updated!')
+
+          logout
+
+          visit shop_path
+        end
+
+        it "filters out variants according to the selected taxon" do
+          expect(page).to have_content variant.name.to_s
+          expect(page).to have_content variant2.name.to_s
+
+          within "#shop-tabs .taxon-selectors" do
+            expect(page).to have_content "Delicious Dandelion"
+            toggle_filter "Delicious Dandelion"
+          end
+
+          expect(page).to have_content variant2.name.to_s
+          expect(page).not_to have_content variant.name.to_s
+        end
+
+        it "filters out variants according to the selected property" do
+          expect(page).to have_content variant.name.to_s
+          expect(page).to have_content variant2.name.to_s
+
+          within "#shop-tabs .property-selectors" do
+            expect(page).to have_content "Fresh and Fine"
+            # toggles filter "Fresh and Fine"
+            page.find(:xpath, '//*[@id="shop-tabs"]/div[2]/div[2]/form/products/div[2]/div/div/div[3]/div[2]/ul/li[1]/a/span').click
```

Comment:
This is terrible practice, but I've sweated quite a lot for this _not to be_ flaky... No idea why but I could not select the `property` filters using the same approach as with the taxon filters, i.e., by simply using the `toggle_filter` helper.... With the precise xpath it just never fails.... Any idea why?

Reply:
Found it. This is because there is more than one div with `#shop-tabs .property-selectors` selector:

<img width="1021" alt="Capture d’écran 2022-04-13 à 11 10 25" src="https://user-images.githubusercontent.com/296452/163142229-902ca424-d39f-45c4-bf32-1a3267137316.png">

Each product has its own `#shop-tabs .property-selectors` which is not clickable. 

You have to be more specific in your `within` block, to be sure to select the content on the right column, by using for example: 

`#shop-tabs .sticky-shop-filters-container .property-selectors`

Reply:
Thanks for the help, makes total sense...

I was quite confused with the outcome from my attempts, as they were all providing a valid result, just not the one I was after. On the byebug console:
```
(byebug) toggle_filter "Fresh and Fine"
#<Capybara::Node::Element:0x00007fe7873fc250>
```

Basically, this was sending clicks to the non-clickable filter, below the product name...
Also, I gather the search function on the console can be really useful. Thanks again.